test/analysis/unittest/interpolator.py

```diff
@@ -0,0 +1,122 @@
+import pytest
+import numpy as np
+
+
+class MockData:
+    def __init__(self, start):
+        pass
+
+
+def linear_data(times, values):
+    for t, y in zip(times, values):
+        yield (t, y)
+
+
+class MockSubscriber:
+    def __init__(self, data_func):
+        self.data_func = data_func
+        self.topic = b"mock"
+
+    def receive(self):
+        return (self.topic, next(self.data_func))
+
+
+class TimestampInterpolator:
```

Comment:
Why is this class defined here in test? Shouldn't the `PandasInterpolator` be tested here, instead?